src/picongpu/include/plugins/ResourceLog.hpp

```diff
@@ -0,0 +1,243 @@
+                    ("resourceLog.prefix", po::value<std::string>(&outputFilePrefix)->default_value("resourceLog_"),
+                     "Set the filename prefix for output file if a filestream was selected")
+                    ("resourceLog.stream", po::value<std::string>(&streamType)->default_value("file"),
+                     "Output stream [stdout, stderr, file]")
+                    ("resourceLog.properties", po::value<std::vector<std::string> >(&properties)->multitoken(),
+                     "List of properties to log [rank, position, currentStep, cellCount, particleCount]")
+                    ("resourceLog.format", po::value<std::string>(&outputFormat)->default_value("json"),
+                     "Output format of log (pp for pretty print) [json, jsonpp, xml, xmlpp]");
+        }
+
+        void setMappingDescription(MappingDesc *cellDescription)
+        {
+            this->cellDescription = cellDescription;
+        }
+
+    private:
+        uint32_t notifyPeriod;
+
+        void pluginLoad() {
+            if(notifyPeriod != 0) {
+                Environment<>::get().PluginConnector().setNotificationPeriod(this, notifyPeriod);
+
+                // Set default resources to log
+                if (properties.empty()) {
+                    properties.push_back("rank");
+                    properties.push_back("position");
+                    properties.push_back("currentStep");
+                    properties.push_back("cellCount");
+                    properties.push_back("particleCount");
+                    propertyMap["rank"] = true;
+                    propertyMap["position"] = true;
+                    propertyMap["currentStep"] = true;
+                    propertyMap["particleCount"] = true;
+                    propertyMap["cellCount"] = true;
+                }
+                else {
+                    for (size_t i = 0; i < properties.size(); ++i) {
+                        propertyMap[properties[i]] = true;
+                    }
+                }
+
+                // Prepare file for output stream
+                if (streamType == "file") {
+                    size_t rank = static_cast<size_t>(Environment<simDim>::get().GridController().getGlobalRank());
+                    std::stringstream ss;
+                    ss << outputFilePrefix << rank;
+                    boost::filesystem::path resourceLogPath(ss.str());
+                    fileBuf.open(resourceLogPath.string().c_str(), std::ios::out);
+                }
+            }
+        }
+
+        void pluginUnload()
+        {
+            if(fileBuf.is_open()){
+                fileBuf.close();
+            }
+            /* called when plugin is unloaded, cleanup here */
+        }
+
+        template <typename T_MAP>
+        bool contains(T_MAP const map, std::string const value)
+        {
+            return (map.find(value) != map.end());
+        }
+
+    };
+}
+
+#include "mappings/simulation/ResourceMonitor.tpp"
```

Comment:
Is this file already part of the `dev`? Why are there no commits to this file in this pull request (or why did we have a `*.tpp` file without this file)?